code/modules/projectiles/projectile_parent.dm

```diff
@@ -375,11 +375,15 @@
 		src.was_setup = 1
 
 	Bump(var/atom/A)
+		if (proj_data?.on_pre_hit(A, src.angle, src))
+			return // Our bullet doesnt want to hit this
```

Comment:
was better to have this in `collide()` I think. Might be worth looking at using/improving the projectile collision signals for this, as they do a *kinda* similar thing for the sniper wallpierce already

Reply:
I thought about using a pierce component, but then I'd likely need to make another one, since I need something that doesn't pierce mobs, but pierces anything else that isnt opaque, and it felt kinda silly to have another component that's basically the same component, but with slightly different parameters.

Though I'll put the proc check back into collide.

Reply:
@Superlagg reminder to do this